### DISCUSSION TITLE
AFNI functions that depend on FreeSurfer

### DISCUSSION THREAD

**Discussion by egoddrd** [neurodesk/neurodesk.github.io#101] [Q&A] [answered]
I'm new to NeuroDesk (and think it's great so far!), and I'm using NeuroDesktop at the moment but might need HPC in future. I'm mostly using AFNI and FreeSurfer, which run fine separately in NeuroDesktop, but there is one afni function, @SUMA_Make_Spec_FS (see https://afni.nimh.nih.gov/pub/dist/doc/htmldoc/tutorials/fatcat_prep/Preprocessing_FS.html), which requires both FreeSurfer and afni. Is it possible to load functionality of two containers simultaneously? Or would it be possible to add FreeSurfer to the afni container? Thanks.

**Comment by stebo85** [comment, marked as answer]
Dear @egoddrd,

Thank you for getting in touch :) I wasn't aware that freesurfer's mris_convert is needed inside AFNI and will add this now to the AFNI container 

Thank you
Steffen

**Comment by stebo85** [comment]
Dear @egoddrd,

I added the functionality to a new AFNI container in version 22.3.07

Could you test quickly if this works for you on here:
https://bhnam.neurodesk.org/

This is what I tested, but not sure if this covers your full usecase:

```
cp /opt/freesurfer-7.3.2/subjects/bert ~/bert -r

\@SUMA_Make_Spec_FS -NIFTI -fspath ~/bert/surf/ -sid bert
```

if this works, I can push a new release of neurodesktop or you can use the latest daily build

Thank you
Steffen

**Reply by egoddrd** [reply]
Hi Steffen, 
\@SUMA_Make_Spec_FS worked for me on the link you sent. 
Thanks very much, 
Erin

**Reply by stebo85** [reply]
Great - I prepare a new release then :) Thank you so much for finding this bug!

**Reply by stebo85** [reply]
Dear @egoddrd - just letting you know that is update is now available in the latest release of neurodesktop: 20221216

**Reply by egoddrd** [reply]
Thanks :)
